src/core/document.rs

```diff
@@ -287,18 +381,21 @@ pub fn print(sd: ServiceDocument) -> String {
                 TypeKind::Union(_) => unions.push(print_type_def(&type_def.node)),
                 TypeKind::InputObject(_) => inputs.push(print_type_def(&type_def.node)),
             },
-            TypeSystemDefinition::Directive(_) => todo!("Directives are not supported yet"),
+            TypeSystemDefinition::Directive(type_def) => {
+                directives.push(print_directive_type_def(&type_def.node))
+            }
         }
     }
 
     // Concatenate the definitions in the desired order
     let sdl_string = schemas
         .into_iter()
-        .chain(scalars)
+        .chain(directives)
         .chain(inputs)
+        .chain(enums)
+        .chain(scalars)
         .chain(interfaces)
         .chain(unions)
-        .chain(enums)
```

Comment:
Do we need to change the order here? Its producing unnecessary diffs, perhaps can be done in a separate PR.

Reply:
We don't really need to change the order, it's not that important I changed them just to make the final generated file look kinda similar. 

The order is : 

- Directives
- Input
- Enums
- Scalar

I'm gonna bring back the old order, if you feel like changing them we can make another PR. 

note: there's some snapshots that got updated because of the new printing format of comments, before the formatting was kinda off.